source/DSCResources/DSC_WebVirtualDirectory/DSC_WebVirtualDirectory.psm1

```diff
@@ -46,11 +50,30 @@ function Get-TargetResource
 
     $PhysicalPath = ''
     $Ensure = 'Absent'
+    $Credential = $null
 
     if ($virtualDirectory.Count -eq 1)
     {
         $PhysicalPath = $virtualDirectory.PhysicalPath
         $Ensure = 'Present'
+
+        if ($WebApplication.Length -gt 0)
+        {
+            $ItemPath = "IIS:Sites\$Website\$WebApplication\$Name"
+        }
+        else
+        {
+            $ItemPath = "IIS:Sites\$Website\$Name"
+        }
+
+        $userName = (Get-ItemProperty $ItemPath -Name userName).Value
+        if ($userName -ne '')
+        {
+            #$password = (Get-ItemProperty $ItemPath -Name password).Value
+            $password = New-Object System.Security.SecureString # Blank Password
```

Comment:
Should this return the real password instead of a blank one?

Reply:
It is not possible to return a password for a PSCredential object. And if it would be possible it wouldn't be advisable.